src/warnet/utils.py

```diff
@@ -408,3 +417,94 @@ def default_bitcoin_conf_args() -> str:
             conf_args.append(f"-{key}={value}")
 
     return " ".join(conf_args)
+
+
+def create_graph_with_probability(graph_func, params: List, version: str, bitcoin_conf: Optional[str], random_version: bool):
+    kwargs = {}
+    for param in params:
+        try:
+            key, value = param.split("=")
+            kwargs[key] = value
+        except ValueError:
+            msg = f"Invalid parameter format: {param}"
+            logger.error(msg)
+            return msg
+
+    # Attempt to convert numerical values from string to their respective numerical types
+    for key in kwargs:
+        try:
+            kwargs[key] = int(kwargs[key])
+        except ValueError:
+            try:
+                kwargs[key] = float(kwargs[key])
+            except ValueError:
+                pass
+
+    logger.debug(f"Parsed params: {kwargs}")
+
+    try:
```

Comment:
IMO most of the try/catch you have in this function should be removed so the error gets thrown back to the RPC caller. Otherwise you get a nice log message but all the CLI shows is "server error"

Reply:
Hmmm, this is an _attempt_. I actually don't recall why now I needed it at all. IIRC during some of my testing some graphs were breaking without it. But could have been an artefact of the `nargs` optional cli args thingy (i.e. dodgy input coming from terminal?)

Perhaps if we want to actually return exceptions back to the user via CLI (do we though!?) we should have a refactor PR which moves all try/except into server.py? Each server function wraps function calls with them and can then either returns success, or any bubbled up exception (can also remove our dodgy decorator at the same time?)

Reply:
just execute `warcli graph create` with no arguments.

User gets this:
`Error generating graph: Server error`

Logs get all this. Which starts off making sense (missing an arg) but ends up in a really weird place...
```
2023-11-01 17:42:21,811 - utils - ERROR - Failed to create graph: random_internet_as_graph() missing 1 required positional argument: 'n'
2023-11-01 17:42:21,846 - warnet.server - ERROR - unexpected error
Traceback (most recent call last):
  File "/opt/homebrew/lib/python3.11/site-packages/flask_jsonrpc/site.py", line 135, in handle_dispatch_except
    return self.dispatch(req_json)
           ^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/lib/python3.11/site-packages/flask_jsonrpc/site.py", line 184, in dispatch
    resp_view = current_app.ensure_sync(view_func)(**params)
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/lib/python3.11/site-packages/typeguard/__init__.py", line 1033, in wrapper
    retval = func(*args, **kwargs)
             ^^^^^^^^^^^^^^^^^^^^^
  File "/Users/matthewzipkin/Desktop/work/warnet/src/warnet/server.py", line 291, in graph_generate
    nx.write_graphml(graph, bio)
  File "/opt/homebrew/lib/python3.11/site-packages/networkx/utils/decorators.py", line 766, in func
    return argmap._lazy_compile(__wrapper)(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<class 'networkx.utils.decorators.argmap'> compilation 9", line 5, in argmap_write_graphml_lxml_5
    import itertools
             ^^^^^^^^
  File "/opt/homebrew/lib/python3.11/site-packages/networkx/readwrite/graphml.py", line 176, in write_graphml_lxml
    writer = GraphMLWriterLxml(
             ^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/lib/python3.11/site-packages/networkx/readwrite/graphml.py", line 745, in __init__
    self.add_graph_element(graph)
  File "/opt/homebrew/lib/python3.11/site-packages/networkx/readwrite/graphml.py", line 751, in add_graph_element
    if G.is_directed():
       ^^^^^^^^^^^^^
AttributeError: 'str' object has no attribute 'is_directed'
```

Reply:
But I hear ya, we might need an RPC overhaul to clean up some `try`'s. I think we are inconsistent with it.